addons/mail/models/mail_rtc_session.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models
+
+
+class MailRtcSession(models.Model):
+    _name = 'mail.rtc.session'
+    _description = 'Mail RTC session'
+
+    channel_id = fields.Many2one('mail.channel', ondelete='cascade', required=True)
+    partner_id = fields.Many2one('res.partner', index=True, required=True, ondelete='cascade')
+
+    is_screen_sharing_on = fields.Boolean()
+    is_camera_on = fields.Boolean()
+    is_muted = fields.Boolean()
+    is_deaf = fields.Boolean()
+
+    def ping(self):
```

Comment:
So this is called every 2mn by every partner currently in a session, and the GC cron runs every 3mn to try and collect sessions, that seems like a lot.

What happens if the cron is disabled and the sessions are not collected?

Reply:
@xmo-odoo Most of the time, sessions are unlinked when the user leaves the call, but if the user doesn't leave the call properly (user loses connection / browser crash / server crash) then the session stays there. It then either removed by the cron, or by the user if they rejoin the same thread (since joining a thread closes other sessions from the same user). In the end there may occasionally be some "dead" sessions in some channels and it only serves at limiting the lifspan of these.

I'll put it to 1 day as noted by odo and reomove the ping

cc @tde-banana-odoo @odony (related comments)